ballerina/tests/test.bal

```diff
@@ -0,0 +1,306 @@
+            },
+            serviceUrl = "https://demo.docusign.net/restapi/"
+        );
+    } else {
+        docuSignClient = check new(
+            {
+                timeout: 10000,
+                auth: {
+                    token: accessToken
+                }
+            },
+            serviceUrl = "http://localhost:9092/restapi"
+        );
+    }
+}
+
+@test:Config {
+    groups: ["account"]
+}
+function testServiceInfo() returns error? {
+    ServiceInformation expectedResponse = {
+        buildVersion:"23.4.0.45303 (23.4.00.00.45303+951e7d1b18c5)",
+        linkedSites:["https://www.docusign.net","https://na2.docusign.net","https://na3.docusign.net","https://eu.docusign.net","https://au.docusign.net","https://ca.docusign.net","https://na4.docusign.net"],
+        serviceVersions: [
+            {
+                version: "v1",
+                versionUrl: "https://demo.docusign.net/restapi/v1"
+            },
+            {
+                version: "v2",
+                versionUrl: "https://demo.docusign.net/restapi/v2"
+            },
+            {   
+                version: "v2.1",
+                versionUrl: "https://demo.docusign.net/restapi/v2.1"
+            }
+        ]
+    };
+    ServiceInformation response = check docuSignClient->/service_information;
+    test:assertEquals(response.serviceVersions, expectedResponse.serviceVersions);
+}
+
+@test:Config {
+    groups: ["account"]
+}
+function testVersion() returns error? {
+    ResourceInformation expectedResponse = {
+        resources: [
+            {
+                name: "accounts",
+                value: "https://demo.docusign.net/restapi/v2.1/accounts"
+            },
+            {
+                name: "billing_plans",
+                value: "https://demo.docusign.net/restapi/v2.1/billing_plans"
+            },
+            {
+                name: "login_information",
+                value: "https://demo.docusign.net/restapi/v2.1/login_information"
+            },
+            {
+                name: "oauth2",
+                value: "https://demo.docusign.net/restapi/v2.1/oauth2"
+            }
+        ]
+    };
+    ResourceInformation response = check docuSignClient->/["v2.1"];
+    test:assertEquals(response.resources, expectedResponse.resources);
+}
+
+@test:Config {
+    groups: ["account"]
+}
+function testCreateAndDeleteUserSignatures() returns error? {
+    string base64Encoded = array:toBase64(check io:fileReadBytes("tests/resources/signature.png"));
+    UserSignaturesInformation payload = {
+        userSignatures: [
+            {
+                imageBase64: base64Encoded,
+                signatureName: "test signature"
+            }
+        ]
+    };
+    UserSignaturesInformation response = check docuSignClient->/accounts/[accountId]/users/[userId]/signatures.post(payload);
+    test:assertEquals((<UserSignature[]>response.userSignatures)[0].signatureName, (<UserSignature[]>payload.userSignatures)[0].signatureName);

```

Comment:
Why do we have to cast here?

Reply:
Fixed in [2f3f735](https://github.com/ballerina-platform/module-ballerinax-docusign.dsesign/pull/1/commits/2f3f7359ffce1ebcaf1d5d76be5eb0b050142d8e)